.eslintrc.js

```diff
@@ -1,44 +1,67 @@
+/* global module */
 module.exports = {
-  root: true,
-  parserOptions: {
-    sourceType: 'module',
+  ignorePatterns: [
+    "!.prettierrc.js",
+    "**/!.eslintrc.js",
+    "**/dist*/",
+    "**/*__GENERATED__*",
+    "**/build",
+    "**/public",
+    "**/.cache",
+    "**/styles",
+  ],
+  parser: "@typescript-eslint/parser",
+  plugins: [
+    "@typescript-eslint",
+    "prettier",
+    // We enforce certain rules on how imports are handled
+    "import",
+  ],
+  extends: [
+    "eslint:recommended",
+    "plugin:prettier/recommended",
+    "plugin:import/errors",
+    "plugin:import/warnings",
+    "plugin:import/typescript",
+  ],
+  rules: {
+    "no-self-compare": "error",
+    "import/order": [
+      1,
+      {
+        groups: [
+          "builtin",
+          "external",
+          "internal",
+          "parent",
+          "sibling",
+          "index",
+        ],
+        "newlines-between": "always",
+      },
+    ],
+    /**
+     * The default setting for Prettier is 'warn' because then it shows as yellow squiggly lines
+     * in the VS Code IDE. However, it means `eslint` will not have an error code if there is warning
+     * due to prettier unles you also add the `--max-warnings=0` flag in front of it. So, in the `lint-staged`
+     * scripts in the packages within this monorepo, we add that flag so that the precommit hooks
+     * associated with that script will fail when run.
+     */
+    "prettier/prettier": "warn",
   },
-
-  extends: ['@metamask/eslint-config'],
-
   overrides: [
     {
-      files: ['**/*.js'],
-      extends: ['@metamask/eslint-config-nodejs'],
-    },
-
-    {
-      files: ['**/*.{ts,tsx}'],
-      extends: ['@metamask/eslint-config-typescript'],
+      files: ["**/*.ts"],
+      extends: ["plugin:@typescript-eslint/recommended"],
       rules: {
-        '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
+        "@typescript-eslint/no-explicit-any": "off",
       },
     },
-
     {
-      files: ['**/*.test.ts', '**/*.test.js'],
-      extends: ['@metamask/eslint-config-jest'],
+      files: ["./src/**/*.ts"],
       rules: {
-        '@typescript-eslint/no-shadow': [
-          'error',
-          { allow: ['describe', 'expect', 'it'] },
-        ],
+        "@typescript-eslint/no-explicit-any": "warn",
```

Comment:
Should we try `error` for this setting and see if we can work with the stricter lint rules?

Reply:
https://github.com/statechannels/nitro-gui/pull/19/commits/db5898edf20411442fd3c6d65cd6fd580d346218